### PR TITLE
[NEW] feat(CAS): Adding option to enable/disable user creation from CAS auth

### LINKED ISSUE #17153: CAS user creation as optional
As feature request:
We would like to make optional the user creation from CAS auth, it's for the case users are created before from a LDAP.
Pull Request is coming.

### PR DESCRIPTION
**Pull request by jgribonvald**
Closes #17153 and [feature request](https://github.com/RocketChat/feature-requests/issues/41)

As simple as it is, this feature permit to make optional the user creation when a user come from a CAS authentication. The use case is that the user is imported before from on other source and so should exist. It permit also to filter user access.

### CODE REVIEW

**Comment by jgribonvald** [comment]
Hi, 
what do you need more to accept this Pull Request ?
Best regards.

**Comment by jgribonvald** [comment]
A little refresh on this PR! What do you think about it ?

**Comment by bseclier** [comment]
Agreed, it would be very useful !

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -624,6 +624,8 @@
   "CAS_button_color": "Login Button Background Color",
   "CAS_button_label_color": "Login Button Text Color",
   "CAS_button_label_text": "Login Button Label",
+  "CAS_Creation_User_Enabled": "Allow user creation",
+  "CAS_Creation_User_Enabled_Description": "Allow CAS User creation from datas provided by the CAS ticket.",
```
Comment (pierre-lehnen-rc) [inline packages/rocketchat-i18n/i18n/en.i18n.json:628]:
```suggestion
  "CAS_Creation_User_Enabled_Description": "Allow CAS User creation from data provided by the CAS ticket.",
```

File: app/cas/server/cas_rocketchat.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ Meteor.startup(function() {
 		this.add('CAS_login_url', '', { type: 'string', group: 'CAS', public: true });
 		this.add('CAS_version', '1.0', { type: 'select', values: [{ key: '1.0', i18nLabel: '1.0' }, { key: '2.0', i18nLabel: '2.0' }], group: 'CAS' });
 		this.add('CAS_trust_username', false, { type: 'boolean', group: 'CAS', public: true, i18nDescription: 'CAS_trust_username_description' });
+		// Enable/disable user creation
+		this.add('CAS_Creation_User_Enabled', true, { type: 'boolean' });
```
Comment (pierre-lehnen-rc) [inline app/cas/server/cas_rocketchat.js:17]:
```suggestion
		this.add('CAS_Creation_User_Enabled', true, { type: 'boolean', group: 'CAS' });
```

File: app/cas/server/cas_server.js [hunk 3]
```diff
@@ -126,6 +126,7 @@ Accounts.registerLoginHandler(function(options) {
 	const sync_enabled = settings.get('CAS_Sync_User_Data_Enabled');
 	const trustUsername = settings.get('CAS_trust_username');
 	const verified = settings.get('Accounts_Verify_Email_For_External_Accounts');
+	const user_creation_enabled = settings.get('CAS_Creation_User_Enabled');
```
Comment (pierre-lehnen-rc) [inline app/cas/server/cas_server.js:129]:
```suggestion
	const userCreationEnabled = settings.get('CAS_Creation_User_Enabled');
```

File: app/cas/server/cas_server.js [hunk 4]
```diff
@@ -205,7 +206,7 @@ Accounts.registerLoginHandler(function(options) {
 				Meteor.users.update(user, { $set: { emails: [{ address: int_attrs.email, verified }] } });
 			}
 		}
-	} else {
+	} else if (user_creation_enabled) {
```
Comment (pierre-lehnen-rc) [inline app/cas/server/cas_server.js:209]:
```suggestion
	} else if (userCreationEnabled) {
```

**Comment by jgribonvald** [comment]
All requested change were applied !

**Comment by jgribonvald** [comment]
Is this PR could be merged now ? as it pass all test and it was approved ?

**Comment by pierre-lehnen-rc** [comment]
Sorry we somehow missed it when cutting the release. I'll make sure it gets merged as soon as the current feature freeze is over.

**Pull request merged**